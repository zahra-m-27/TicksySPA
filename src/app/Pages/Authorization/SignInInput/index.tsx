import API from '../../../API';
import {Button, message} from 'antd';
import {useRef, useState} from 'react';
import styles from './styles.module.scss';
import useUser from '../../../Hooks/useUser';
import SEInput from '../../../Components/SEInput';
import {Link, useHistory} from 'react-router-dom';
import TButton from '../../../Components/TButton';

interface Props {
  className?: string;
}

export default function SignInInput({className}: Props) {
  const {Login} = useUser();
  const history = useHistory();
  const [Email, setEmail] = useState('');
  const [Password, setPassword] = useState('');
  const [Loading, setLoading] = useState(false);
  const passwordRef = useRef<HTMLInputElement>(null);
  const [EmailHasError, setEmailHasError] = useState(false);
  const [PasswordHasError, setPasswordHasError] = useState(false);

  const onEmailEnter = () => {
    if (passwordRef && passwordRef.current) {
      passwordRef.current.focus();
    }
  };

  const onSignIn = () => {
    if (!/^\S+@\S+$/.test(Email)) {
      setEmailHasError(true);
    }
    if (!Password) {
      setPasswordHasError(true);
    }
    if (!Password || !/^\S+@\S+$/.test(Email)) {
      return;
    }
    setEmailHasError(false);
    setPasswordHasError(false);

    setLoading(true);
    API.Users.SignIn({
      username: Email,
      password: Password,
    })
      .then((response) => {
        localStorage.setItem('token', response.token);
        API.Users.GetProfile({}).then((response) => {
          Login(response);
          setLoading(false);
          if (history.location.pathname === '/sign-in') {
            history.push('/');
          }
        });
      })
      .catch((error) => {
        if (error.status === 400) {
          message.error('ایمیل یا رمزعبور اشتباه است');
        }
        setLoading(false);
      });
  };

  return (
    <form className={className}>
      <p className={styles.input_box_title}>وارد شوید!</p>
      <SEInput
        type="text"
        label="ايميل"
        name="email"
        regex={/^\S+@\S+$/}
        onEnter={onEmailEnter}
        onChangeText={setEmail}
        hasError={EmailHasError}
        hint="example@gmail.com"
        data-testid="email-input"
        inputClassName={styles.input}
        className={styles.input_class}
        labelClassName={styles.input_label}
        innerContainerClassName={styles.inner_container}
      />
      <SEInput
        name="password"
        type="password"
        label="گذرواژه"
        ref={passwordRef}
        onEnter={onSignIn}
        onChangeText={setPassword}
        hasError={PasswordHasError}
        data-testid="password-input"
        inputClassName={styles.input}
        className={styles.input_class}
        labelClassName={styles.input_label}
        innerContainerClassName={styles.inner_container}
      />
      <TButton
        label="ورود"
        onClick={onSignIn}
        data-testid="submit-button"
        className={styles.enter_button}
      />
      <Link className={styles.switch} to="/sign-up">
        حساب کابری ندارم
      </Link>
      <Link
        className={styles.forget_pass}
        to="/forgot-password"
        data-testid="forgot-button">
        فراموشی گذرواژه
      </Link>
    </form>
  );
}
