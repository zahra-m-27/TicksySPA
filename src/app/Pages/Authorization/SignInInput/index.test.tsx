import ReactDOM from 'react-dom';
import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import SignInInput from './index';
import {MemoryRouter} from 'react-router-dom';

const sample = (
  <MemoryRouter>
    <SignInInput />
  </MemoryRouter>
);

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render title', () => {
  const {getByText} = render(sample);
  getByText(/وارد شوید/i);
});

it('render email field', () => {
  const {getAllByText} = render(sample);
  getAllByText(/ايميل/i);
});

it('render password field', () => {
  const {getAllByText} = render(sample);
  getAllByText(/گذرواژه/i);
});

it('render submit button', () => {
  const {getByText} = render(sample);
  getByText(/ورود/i);
});

it('render sign up button', () => {
  const {getByText} = render(sample);
  getByText(/حساب کابری ندارم/i);
});

it('render forgot button', () => {
  const {getByText} = render(sample);
  getByText(/فراموشی گذرواژه/i);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
