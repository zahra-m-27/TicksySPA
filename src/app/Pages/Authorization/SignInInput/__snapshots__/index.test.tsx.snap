// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<form>
  <p
    className="input_box_title"
  >
    وارد شوید!
  </p>
  <div
    className="input_container input_class"
  >
    <div
      className="inner_container inner_container"
    >
      <div
        className="content"
      >
        <p
          className="label input_label"
        >
          ايميل
        </p>
        <input
          className="input input input_label"
          data-testid="email-input"
          dir="auto"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="input_container input_class"
  >
    <div
      className="inner_container inner_container"
    >
      <div
        className="content"
      >
        <p
          className="label input_label"
        >
          گذرواژه
        </p>
        <input
          className="input input input_label"
          data-testid="password-input"
          dir="auto"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          type="password"
          value=""
        />
        <div
          data-testid="password-visibility"
          onClick={[Function]}
        >
          <svg
            className="eye_svg"
            fill="#5B5B5B"
            height={24}
            viewBox="0 0 16 16"
            width={24}
          >
            <path
              d="M7.5 4C2.105469 4 0.046875 8.289063 0.046875 8.289063L-0.0546875 8.5L0.046875 8.710938C0.046875 8.710938 2.105469 13 7.5 13C12.894531 13 14.953125 8.710938 14.953125 8.710938L15.050781 8.5L14.953125 8.289063C14.953125 8.289063 12.894531 4 7.5 4 Z M 7.5 5C9.4375 5 11 6.5625 11 8.5C11 10.4375 9.4375 12 7.5 12C5.5625 12 4 10.4375 4 8.5C4 6.5625 5.5625 5 7.5 5 Z M 3.941406 5.777344C3.359375 6.535156 3 7.472656 3 8.5C3 9.527344 3.359375 10.464844 3.941406 11.222656C2.023438 10.269531 1.242188 8.753906 1.117188 8.5C1.242188 8.246094 2.023438 6.730469 3.941406 5.777344 Z M 11.058594 5.777344C12.976563 6.730469 13.757813 8.246094 13.882813 8.5C13.757813 8.753906 12.976563 10.269531 11.058594 11.222656C11.640625 10.464844 12 9.527344 12 8.5C12 7.472656 11.640625 6.535156 11.058594 5.777344 Z M 7.5 7C6.671875 7 6 7.671875 6 8.5C6 9.328125 6.671875 10 7.5 10C8.328125 10 9 9.328125 9 8.5C9 7.671875 8.328125 7 7.5 7Z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <button
    className="ant-btn ant-btn-primary enter_button"
    data-testid="submit-button"
    onClick={[Function]}
    type="button"
  >
    <span>
      ورود
    </span>
  </button>
  <a
    className="switch"
    href="/sign-up"
    onClick={[Function]}
  >
    حساب کابری ندارم
  </a>
  <a
    className="forget_pass"
    href="/forgot-password"
    onClick={[Function]}
  >
    فراموشی گذرواژه
  </a>
</form>
`;
